src/stp/new_plays/Attack.cpp

```diff
@@ -2,13 +2,18 @@
 // Created by jordi on 24-03-20.
 //
 
+#include "stp/invariants/WeHaveBallInvariant.h"
 #include "stp/new_plays/Attack.h"
 #include "stp/new_roles/Attacker.h"
 #include "stp/new_roles/TestRole.h"
 
 namespace rtt::ai::stp::play {
 
 Attack::Attack(std::string playName) : Play(playName) {
+    // TODO: decide invariants
+    invariants.clear();
+    invariants.emplace_back(std::make_unique<invariant::WeHaveBallInvariant>());
```

Comment:
Maybe use the same structure as constructing the roles i.e. `invariants = std::vector<?>{std::make_unique<invariant_type>()}`